bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableTable.java

```diff
@@ -274,22 +274,19 @@ private FlatRow getResults(Get get, String method) {
     }
   }
 
-  protected Result convertToResult(FlatRow row) {
-    return Adapters.FLAT_ROW_ADAPTER.adaptResponse(row);
-  }
-
   /** {@inheritDoc} */
   @Override
-  public ResultScanner getScanner(Scan scan) throws IOException {
+  public ResultScanner getScanner(final Scan scan) throws IOException {
     LOG.trace("getScanner(Scan)");
     Span span = TRACER.spanBuilder("BigtableTable.scan").startSpan();
     try (Scope scope = TRACER.withSpan(span)) {
-      com.google.cloud.bigtable.grpc.scanner.ResultScanner<FlatRow> scanner =
-          clientWrapper.readFlatRows(hbaseAdapter.adapt(scan));
+
+      final ResultScanner scanner = clientWrapper.readRows(hbaseAdapter.adapt(scan));
       if (hasWhileMatchFilter(scan.getFilter())) {
         return Adapters.BIGTABLE_WHILE_MATCH_RESULT_RESULT_SCAN_ADAPTER.adapt(scanner, span);
       }
-      return Adapters.BIGTABLE_RESULT_SCAN_ADAPTER.adapt(scanner, span);
+      // TODO: need to end the span when stream ends
```

Comment:
😞 Thanks for noticing this. Can you open github issue for this so it doesnt get lost.
I think a good fix for this is to extract trace management to a separate Scanner wrapper

Reply:
Thanks for the review, Have created #2458 to track the same.